cmd/flux/main_test.go

```diff
@@ -298,6 +302,13 @@ func assertGoldenTemplateFile(goldenFile string, templateValues map[string]strin
 			expectedOutput = string(goldenFileContents)
 		}
 		if assertErr := assertGoldenValue(expectedOutput)(output, err); assertErr != nil {
+			// Update the golden files if comparision fails and the update flag is set.
+			if *update && output != "" {
+				if err := os.WriteFile(goldenFile, []byte(output), 0644); err != nil {
```

Comment:
Wouldn't this remove the variables declared inside golden files such as https://github.com/fluxcd/flux2/blob/437a7a2852976c037d34915d325d9fb5288ebd77/cmd/flux/testdata/trace/deployment.golden#L9

Reply:
Yeah, that'll update the golden template files as well. Didn't realize that the golden files are in template form.
After going through the code again, it looks like we can determine when it's a golden template file based on the `templateValues` passed to `assertGoldenTemplateFile()`.
I've added a check for that. Now when there's a diff in the output **and** `-update` is passed, a note is printed about it.

```console
NOTE: -update flag passed but golden template file can't be updated, please update it manually
    main_test.go:351: Mismatch from golden file 'testdata/trace/helmrelease.golden': Mismatch from expected value (-want +got):
```

Since not all the packages use golden files and won't work with `-update` flag, I made some changes in the Makefile to make it easy to set the target test package and documented how to do it in the CONTRIBUTING.md file.
Does it look good?

Reply:
Great! thank you